packages/api/src/external/carequality/command/run-or-schedule-patient-discovery.ts

```diff
@@ -0,0 +1,49 @@
+import { Patient } from "@metriport/core/domain/patient";
+import { getPatientOrFail } from "../../../command/medical/patient/get-patient";
+import { MedicalDataSource } from "@metriport/core/external/index";
+import { schedulePatientDiscovery } from "../../hie/schedule-patient-discovery";
+import { getCQData, discover } from "../patient";
+
+export async function runOrScheduleCqPatientDiscovery({
+  patient,
+  facilityId,
+  requestId,
+  rerunPdOnNewDemographics,
+  forceCarequality,
+}: {
+  patient: Patient;
+  facilityId: string;
+  requestId: string;
+  rerunPdOnNewDemographics?: boolean;
+  // START TODO #1572 - remove
+  forceCarequality?: boolean;
+  // END TODO #1572 - remove
+}): Promise<void> {
+  const existingPatient = await getPatientOrFail({
+    id: patient.id,
+    cxId: patient.cxId,
+  });
+  const cqData = getCQData(patient.data.externalData);
+
+  const discoveryStatusCq = cqData?.discoveryStatus;
+  const scheduledPdRequestCq = cqData?.scheduledPdRequest;
+
+  if (discoveryStatusCq === "processing" && !scheduledPdRequestCq) {
```

Comment:
What if a cx changes a patient 3 times all with different demographics? 
- First will run and set to processing 
- Second will be scheduled 
- We lose the 3rd one because we do nothing 

Theres a pretty small chance this scenario ever happens but why wouldnt we just add all updates to a queue?

Reply:
May not be worth addressing now because it seems like we only have the capability to schedule one thing at a time but just something to keep in mind

Reply:
The second one that's scheduled, when it runs, will reflect the updates from the third one, so we don't need the third one. The thing we lose would be is the user wanted to run the third one with `rerunPdOnNewDemopgraphics` -- in which case this param wouldn't update the scheduled PD.

We could overwrite the scheduled PD if we wanted. Either behavior is fine (i.e. first one that's scheduled stays, or overwrite the scheduled PD on each update).

Reply:
@Orta21 that work for you?